### DISCUSSION TITLE
I'd like 'submodules' in Dark

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4377] [Suggestions]
Some modules (esp. `DarkInternal`) are rather flat - it'd be great to break them down into submodules.

For example, I'd like my code to look like `DarkInternal.StaticAssets.startDeploy` rather than `DarkInternal.startStaticAssetDeploy`.
I haven't thought too deeply about this - just opening the topic for conversation.

**Comment by StachuDotNet** [comment]
Some quick thoughts on implementation:
`fn "DarkInternal" "startStaticAssetDeploy" 0` becomes `fn "DarkInternal" ["StaticAssets"] "startDeploy" 0`.
(the first paramater is the root module - a new 2nd parameter outlines the nested sub-modules from there. (so that I can define `A.B.C.d` as `fn "A" ["B"; "C"] "d" 0`

Similarly, `FQFnName` and probably other components would have to be updated in turn.

**Comment by pbiggar** [comment]
Users have also expressed wanting to create modules for their functions. They'll also be needed for types and packages.

A nice thing about modules versus extending the function name, is that we can hide modules by default in the code layout, only showing modules in the presence of ambiguity or user setting.

Eg: `append` doesn't always need `List::` attached to it, but if we're working on dicts as well as lists, then we should make clear which you're looking at. (The is sort of like how people use imports in many existing languages, except this is automated)
